tests/integration/test_charm.py

```diff
@@ -0,0 +1,269 @@
+
+    # test connection for PostgreSQL with retrieved credentials
+    with psycopg2.connect(connection_string) as connection, connection.cursor() as cursor:
+        # Check that it's possible to write and read data from the database that
+        # was created for the application.
+        connection.autocommit = True
+        cursor.execute(f"DROP TABLE IF EXISTS {DATABASE_NAME};")
+        cursor.execute(f"CREATE TABLE {DATABASE_NAME}(data TEXT);")
+        cursor.execute(f"INSERT INTO {DATABASE_NAME}(data) VALUES('some data');")
+        cursor.execute(f"SELECT data FROM {DATABASE_NAME};")
+        data = cursor.fetchone()
+        assert data[0] == "some data"
+
+        # Check the version that the application received is the same on the database server.
+        cursor.execute("SELECT version();")
+        data = cursor.fetchone()[0].split(" ")[1]
+
+        assert version == data
+
+    #  remove relation and test connection again
+    await ops_test.model.applications[DATA_INTEGRATOR].remove_relation(
+        f"{DATA_INTEGRATOR}:postgresql", f"{POSTGRESQL}:database"
+    )
+
+    await ops_test.model.wait_for_idle(apps=[POSTGRESQL, DATA_INTEGRATOR])
+    await ops_test.model.add_relation(DATA_INTEGRATOR, POSTGRESQL)
+    await ops_test.model.wait_for_idle(apps=[DATA_INTEGRATOR, POSTGRESQL])
+
+    new_credentials = await fetch_action_get_credentials(
+        ops_test.model.applications[DATA_INTEGRATOR].units[0]
+    )
+
+    # Build the complete connection string to connect to the database.
+    new_connection_string = build_postgresql_connection_string(new_credentials)
+    # check that new credentials are provided
+    assert new_connection_string != connection_string
+
+    # Connect to the database using different credentials.
+    with psycopg2.connect(new_connection_string) as connection, connection.cursor() as cursor:
+        # Read data from previously created database.
+        cursor.execute(f"SELECT data FROM {DATABASE_NAME};")
+        data = cursor.fetchone()
+        assert data[0] == "some data"
+
+
+async def test_deploy_and_relate_mongodb(ops_test: OpsTest):
+    """Test the relation with MongoDB and database accessibility."""
+    await asyncio.gather(
+        ops_test.model.deploy(
+            "mongodb", channel="dpe/edge", application_name=MONGODB, num_units=1, series="focal"
+        )
+    )
+    await ops_test.model.wait_for_idle(apps=[MONGODB])
+    assert ops_test.model.applications[MONGODB].status == "active"
+    await ops_test.model.add_relation(DATA_INTEGRATOR, MONGODB)
+    await ops_test.model.wait_for_idle(apps=[DATA_INTEGRATOR, MONGODB])
+    assert ops_test.model.applications[DATA_INTEGRATOR].status == "active"
+
+    # get credential for MongoDB
+    credentials = await fetch_action_get_credentials(
+        ops_test.model.applications[DATA_INTEGRATOR].units[0]
+    )
+
+    connection_string = credentials[MONGODB]["uris"]
+
+    client = MongoClient(
```

Comment:
Should both pymongo and psycopg2 also be into their own context manager classes as well?